src/tiled/mapobjectmodel.cpp

```diff
@@ -43,6 +43,8 @@ MapObjectModel::MapObjectModel(QObject *parent):
     mObjectGroupIcon(QLatin1String(":/images/16x16/layer-object.png"))
 {
     mObjectGroupIcon.addFile(QLatin1String(":images/32x32/layer-object.png"));
+    connect(this, &MapObjectModel::objectsChanged,
+            this, &MapObjectModel::emitObjectsDataChanged);
```

Comment:
Instead of connecting this signal, I'd prefer if `emitObjectsDataChanged` was only called when needed, and only with the actually changed columns. The `objectsChanged` signal gets emitted for many things that aren't actually exposed by this model, and `dataChanged` is already emitted for a few things explicitly as well.